FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/ExternalDependencyChecker.cs

```diff
@@ -113,26 +121,44 @@ private bool IsBlobContainerReady()
             return isReady;
         }
 
-        private async Task<bool> IsFhirDataClientReady(CancellationToken cancellationToken)
+        private async Task<bool> IsApiDataClientReady(CancellationToken cancellationToken)
         {
-            // TODO DICOM: health check
-            /*var queryParameters = new List<KeyValuePair<string, string>>
+            BaseApiOptions searchOptions;
+            switch (_dataSourceType)
             {
-                new KeyValuePair<string, string>(FhirApiConstants.PageCountKey, FhirApiPageCount.Single.ToString("d")),
-            };
-            var searchOptions = new BaseSearchOptions("Patient", queryParameters);
+                case DataSourceType.FHIR:
+                    var fhirQueryParameters = new List<KeyValuePair<string, string>>
+                    {
+                        new KeyValuePair<string, string>(FhirApiConstants.PageCountKey, FhirApiPageCount.Single.ToString("d")),
+                    };
+                    searchOptions = new BaseSearchOptions("Patient", fhirQueryParameters);
+                    break;
+                case DataSourceType.DICOM:
+                    var dicomQueryParameters = new List<KeyValuePair<string, string>>
+                    {
+                        new KeyValuePair<string, string>(DicomApiConstants.IncludeMetadataKey, $"{false}"),
+                    };
+                    searchOptions = new ChangeFeedLatestOptions(dicomQueryParameters);
+                    break;
+                default:
+                    // Should not happend.
+                    _logger.LogInformation($"[External Dependency Check] Unsupported data source type: {_dataSourceType}");
+                    _diagnosticLogger.LogError($"Unsupported data source type: {_dataSourceType}");
+                    return false;
+            }
+
             try
             {
-                // Ensure we can search from FHIR server.
-                await _fhirApiDataClient.SearchAsync(searchOptions, cancellationToken);
-                _logger.LogInformation("[External Dependency Check] Fhir server is ready.");
+                // Ensure we can search from the source Api server.
+                await _apiDataClient.SearchAsync(searchOptions, cancellationToken);
+                _logger.LogInformation("[External Dependency Check] The source Api server is ready.");
             }
             catch (Exception ex)
             {
-                _logger.LogInformation($"[External Dependency Check] Fhir server is not accessible: {ex.Message}.");
-                _diagnosticLogger.LogError($"Fhir server is not accessible: {ex.Message}.");
+                _logger.LogInformation($"[External Dependency Check] The source Api server is not accessible: {ex.Message}.");
+                _diagnosticLogger.LogError($"The source Api serveris not accessible: {ex.Message}.");
```

Comment:
Nit: add space between `server` and `is`.

Reply:
And maybe capitalize `API`?

Reply:
Updated, thanks.